macOS/DuckDuckGo/TabBar/View/TabBarViewController.swift

```diff
@@ -666,8 +666,17 @@ final class TabBarViewController: NSViewController, TabBarRemoteMessagePresentin
     override func mouseMoved(with event: NSEvent) {
         super.mouseMoved(with: event)
 
+        guard collectionView.isMouseLocationInsideBounds() else {
+            if tabPreviewWindowController.isPresented {
+                hideTabPreview(allowQuickRedisplay: true)
+            }
+            return
+        }
```

Comment:
If the mouse it outside the visible bounds, we check if there's a preview to hide and bail out.

Reply:
this one is not correct, the previews are shown not only for the regular tabs but for pinned tabs as well (see the lines below), so not only the `collectionView` should be inspected for bounds: now the Pinned tab previews got broken.
It would be better to add the 3rd if-else branch into the `mouseMoved` hiding the preview if needed

Reply:
I was trying something similar today, but got stuck with the visibility check not quite working - unclear why.  I'll give it another shot tomorrow morning.

Reply:
Ok, I just rolled back my changes and tried again and it's working well now.  The logic in `self.hideTabPreview(allowQuickRedisplay: true)` apparently needs no change.

The logic that is kinda off is the code to get the index for the selected tab since it doesn't properly account for visibility, but this is now fixed because of the added guard checking for `shouldDisplayTabPreviews`.